Lib/test/test_tkinter/test_widgets.py

```diff
@@ -20,9 +26,14 @@
 def float_round(x):
     return float(round(x))
 
-
 class AbstractToplevelTest(AbstractWidgetTest, PixelSizeTests):
-    _conv_pad_pixels = False
+    _no_round = {'padx', 'pady'} if tk_version < (9, 0) else {
+        'borderwidth', 'height', 'highlightthickness', 'padx', 'pady',
+        'width'}
+    _clipped = {
+        'highlightthickness'} if tk_version < (9, 0) else {
+        'borderwidth', 'height', 'highlightthickness', 'padx', 'pady',
+        'width'}
```

Comment:
This is pretty much a styling nit, but this might be better done with a base set that is extended if tk is too new:

```suggestion
    _no_round = {'padx', 'pady'}
    _clipped = {'highlightthickness'}
    if tk_version >= (9, 0):
        _no_round.update(
            {
                'borderwidth',
                'height',
                'highlightthickness',
                'width',
            }
        )
        _clipped.update(_no_round)
```

Or as a single

```
if tk_version < (9, 0):
    # set stuff
else:
    # set stuff
```

branch.  The two ternaries back-to-back aren't especially readable to me :)

Reply:
I wish I knew whether I could count on _no_round being a subset of _clipped.  I think the cases might be more result at the expense of being more verbose.

Reply:
Addressed in commit 5a954ecffc.